WADF.php

```diff
@@ -398,7 +399,14 @@ public function deployDatabase()
 					mysqli_query($db, "CREATE DATABASE IF NOT EXISTS $name");
 				}
 				if (in_array('grant', $deploy_options)) {
-					mysqli_query($db, "GRANT ALL on $name.* to $user IDENTIFIED BY '$pass'");
+					if ($db->server_version >= 80000) {
+						$user_host = $this->resolveMacro("db${num}_user_host");
+						$this->_debugOutput("\tCreating user '{$user}'@'{$user_host}'");
+						mysqli_query($db, "CREATE USER IF NOT EXISTS '{$user}'@'{$user_host}' IDENTIFIED BY '{$pass}'");
+						mysqli_query($db, "GRANT ALL on {$name}.* to '{$user}'@'{$user_host}'");
+					} else {
+						mysqli_query($db, "GRANT ALL on {$name}.* to {$user} IDENTIFIED BY '{$pass}'");
```

Comment:
this line ought to to apply the new db_user_host to MySQL <8

Reply:
(ideally we wouldn't have separate code paths here for different versions, which would be even better. In that case though, one can't use "CREATE USER IF NOT EXISTS" as that syntax doesn't exist on earlier MySQL versions.

Reply:
I've updated the code to  apply to MySQL < 8.
(I agree on the separate code paths, unfortunately MySQL versions are not backwards compatible on this.)